python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -307,17 +309,27 @@ public void visitImportName(ImportName importName) {
           .toList();
         var resolvedType = projectLevelTypeTable.getType(fqn);
 
-        if (!(resolvedType instanceof ModuleType module)) {
-          return;
-        }
-        if (aliasedName.alias() != null) {
-          setTypeToName(aliasedName.alias(), module);
-        } else {
-          for (int i = names.size() - 1; i >= 0; i--) {
-            setTypeToName(names.get(i), module);
-            module = Optional.ofNullable(module)
-              .map(ModuleType::parent)
-              .orElse(null);
+
+        if (resolvedType instanceof ModuleType module) {
+          if (aliasedName.alias() != null) {
+            setTypeToName(aliasedName.alias(), module);
+          } else {
+            for (int i = names.size() - 1; i >= 0; i--) {
+              setTypeToName(names.get(i), module);
+              module = Optional.ofNullable(module)
+                .map(ModuleType::parent)
+                .orElse(null);
+            }
+          }
+        } else if (resolvedType instanceof UnknownType || (resolvedType instanceof ObjectType objectType && objectType.type() instanceof UnknownType)) {
+          if (aliasedName.alias() != null) {
+            var unresolvedImportType = new UnresolvedImportType(String.join(".", fqn));
```

Comment:
For this prototype/investigation, it doesn't matter, but when we actually implement this we should probably try to ensure we don't create duplicate types for the same FQN if we see stuff like:
```
def foo():
    from lib import some_func

def bar():
    from lib import some_func 
```